### PR TITLE
[full ci] Integration tests don't ignore vic-machine delete errors

### LINKED ISSUE #2106: Flag vic-machine delete errors in all tests in addition to the targeted ones
We're currently ignoring errors from vic-machine delete except in the case of the specifically targeted tests. This is going to make it much harder to catch/verify #1787 or tell how reliable delete is in general.

It's also misleading in the case of build https://ci.vmware.run/vmware/vic/4295 where everything looks happy until we start seeing the disk space errors.

### PR DESCRIPTION
**Pull request by andrewtchin**
Fixes #2106 

Passed local regression tests

### CODE REVIEW

File: tests/resources/Util.robot [hunk 1]
```diff
@@ -105,8 +105,9 @@ Cleanup VIC Appliance On Test Server
 
 Run VIC Machine Delete Command
     [Tags]  secret
-    ${output}=  Run  bin/vic-machine-linux delete --name=${vch-name} --target=%{TEST_URL} --user=%{TEST_USERNAME} --password=%{TEST_PASSWORD} --force=true --compute-resource=%{TEST_RESOURCE} --timeout %{TEST_TIMEOUT}
-    Run Keyword And Ignore Error  Should Contain  ${output}  Completed successfully
+    ${rc}  ${output}=  Run And Return Rc And Output  bin/vic-machine-linux delete --name=${vch-name} --target=%{TEST_URL} --user=%{TEST_USERNAME} --password=%{TEST_PASSWORD} --force=true --compute-resource=%{TEST_RESOURCE} --timeout %{TEST_TIMEOUT}
+    Should Be Equal As Integers  ${rc}  0
+    Should Contain  ${output}  Completed successfully
```
Comment (mhagen-vmware) [inline tests/resources/Util.robot]: I also added the following as a start to this before I got distracted this past several weeks with nimbus business:

    ${out}=  Run  govc ls vm
    Log  ${out}
    ${out}=  Run  govc datastore.ls
    Log  ${out}
    ${out}=  Run  govc ls host/*/Resources/*
    Log  ${out}

I think we really need to also add a final check to make sure everything got cleaned up something to the effect of:
    ${out}=  Run  govc datastore.ls
    Should not contain  ${out}  ${vch-name}

And do that for vms and pools as well

Comment (mhagen-vmware) [inline tests/resources/Util.robot]: I was also considering some sort of differential logic as well.  Like check before the delete happens and after and diff them and make sure it is as expected.

Comment (andrewtchin) [inline tests/resources/Util.robot]: added the things from the first comment

**Comment by mhagen-vmware** [comment]
lgtm

**Comment by emlin** [comment]
lgtm 👍

**Comment by andrewtchin** [comment]
♻️

**Pull request merged**